File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    s_list = list(s)
    
    for i in range(len(s_list)):
        move = ord(s_list[i])+n
        
        if s_list[i] == " ":
            answer += " "
        elif 65 <= move and move <= 90 and s_list[i].isupper():
            answer += chr(move)
        elif 97 <= move and move <= 122 and s_list[i].islower():
            answer += chr(move)
        else:
            answer += chr(move - 26)
    
    return answer

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        char[] sArr = s.toCharArray();
        char ch = 'A';
        
        for(int i=0; i<sArr.length; i++) {
            if(sArr[i] == ' ') {
                answer += " ";
            } else {
                ch = (int)sArr[i] >= 97 ? 'a' : 'A';
                answer += (char)(ch + (sArr[i] + n - ch) % 26);
            }
        }
        return answer;
    }
}